### DISCUSSION TITLE
can't create fcm branch from existing one with --branch-of-branch option

### DISCUSSION THREAD

**Discussion by jonnyhtw** [MetOffice/simulation-systems#92] [UM]
hi there, 

apologies if this isn't the right place to post this but i thought i'd give it a go!

there is a pre-existing branch on MOSRS...

```
> fcm info fcm:nemo.x/branches/UKMO/dev_r5518_GO6_package_antony_isf_fix@16043
Path: dev_r5518_GO6_package_antony_isf_fix
URL: https://code.metoffice.gov.uk/svn/nemo/branches/UKMO/dev_r5518_GO6_package_antony_isf_fix
Relative URL: ^/branches/UKMO/dev_r5518_GO6_package_antony_isf_fix
Repository Root: https://code.metoffice.gov.uk/svn/nemo
Repository UUID: 4aad9cc9-4d31-0410-b1e8-ee312aa4b1ec
Revision: 16043
[etc]
```

... but when i try to create a branch from it, i get an error regarding that revision not existing at the trunk...

```
> fcm branch-create niwa fcm:nemo.x/branches/UKMO/dev_r5518_GO6_package_antony_isf_fix@16043 --branch-of-branch
[FAIL] svn info --xml https://code.metoffice.gov.uk/svn/nemo/trunk@16043 # rc=1
[FAIL] svn: warning: W170000: URL 'https://code.metoffice.gov.uk/svn/nemo/trunk' non-existent in revision 16043
[FAIL]
[FAIL] svn: E200009: Could not display info for all targets because some targets don't exist
````

any ideas what i'm doing wrong? 

many thanks,

jonny williams
niwa, nz

**Comment by dpmatthews** [comment]
Unfortunately this repository wasn't set up for use with fcm and doesn't follow the structure expected by fcm. Therefore, although most fcm commands will work, some (especially anything to do with branches or merging) will not.

**Reply by HengistPodd** [reply]
Hi Jonny, If you need to create a branch, you'll need to do it directly on the IPSL repository, though in this case, you could check out the branch from SRS and just make local changes to a working copy, if that suites your purposes. 
(Richard H)

**Reply by jonnyhtw** [reply]
> Unfortunately this repository wasn't set up for use with fcm and doesn't follow the structure expected by fcm. Therefore, although most fcm commands will work, some (especially anything to do with branches or merging) will not.

Thanks @dpmatthews, makes sense! I'm pleased I got the fcm syntax right at least!

**Reply by jonnyhtw** [reply]
> Hi Jonny, If you need to create a branch, you'll need to do it directly on the IPSL repository, though in this case, you could check out the branch from SRS and just make local changes to a working copy, if that suites your purposes. 
> 
> (Richard H)

Thanks Richard, roger that! The old IPSL repo is read only now though isn't it? Or are you referring to the new GitHub one? For the purposes of what I wanted to do in this example I checked out the code and made local changes as you said.

**Reply by HengistPodd** [reply]
Hi Jonny, 
Yes, good point! The genuine "old IPSL repo" is indeed read only now, but we have a copy of it which is writeable on the MO MOSRS system for pre vn4.0 code branches. See: https://code.metoffice.gov.uk/trac/nemo.

So (assuming you have access to MOSRS) you should be able to create branches and do what you need to do on there. 

(As you point out, it's only NEMO vn4.0 onwards which uses Git(Lab).)

Richard

**Reply by jonnyhtw** [reply]
> So (assuming you have access to MOSRS) you should be able to create branches and do what you need to do on there.

thanks @HengistPodd. that's exactly what i was trying to do though using...

```
fcm branch-create niwa fcm:nemo.x/branches/UKMO/dev_r5518_GO6_package_antony_isf_fix@16043 --branch-of-branch
```

if i've understood you correctly then branching *is* possible on `fcm:nemo.x` but @dpmatthews you said that...

> although most fcm commands will work, some (especially anything to do with branches or merging) will not.

i have a working solution now (editing a working copy) but would like to understand what's going on under the hood! 😄

**Reply by dpmatthews** [reply]
You can't use `fcm` because the repository isn't structured correctly. However, in Subversion, branches are really just copies. So, you can still create a "branch" using `svn copy`. Here's a recent example: https://code.metoffice.gov.uk/trac/nemo/changeset/16380
I'm afraid I don't know if there are any branch naming conventions or any other guidance for working with this repository.

**Comment by jonnyhtw** [comment]
> You can't use `fcm` because the repository isn't structured correctly. However, in Subversion, branches are really just copies. So, you can still create a "branch" using `svn copy`. Here's a recent example: https://code.metoffice.gov.uk/trac/nemo/changeset/16380 I'm afraid I don't know if there are any branch naming conventions or any other guidance for working with this repository.

ah yes ok that makes sense (and works too, just tried).

thanks a lot
